### PR TITLE
Fix repeating id form bug

### LINKED ISSUE #74: [BUG] "forms" argument fails when the record_id form is repeating
# Expected Behavior
When a `read_redcap_tidy()` function is called, all expected forms should be returned regardless of `forms` specification and `structure`.

# Current Behavior
When specifying `forms` from a REDCap built where the `record_id` exists in a repeating form, the execution errors.

## How to Reproduce the Bug:
Have a REDCap project where the initial form (the one containing the `record_id` field) is a repeating instrument. Call `read_redcap_tidy` and specify a form that is not that instrument.

**Screenshots**
image

## Failure Logs

Error message:
```
Error in if (my_fields[1] != my_record_id) { : 
missing value where TRUE/FALSE needed
```
Coming from `distill_repeat_table`:
https://github.com/CHOP-CGTDataOps/REDCapTidieR/blob/3944c913990bcd8f9f314a591357c20e01d2adc4/R/clean_redcap.R#L198-L200

### Checklist
Before submitting this issue, please check and verify below that the submission meets the below criteria:

- [x] The issue is atomic
- [x] The issue description is documented
- [x] The issue title describes the problem succinctly
- [x] Developers are assigned to the issue
- [x] Labels are assigned to the issue

### PR DESCRIPTION
**Pull request by ezraporter**
# Description
This PR resolves the issue where `read_redcap_tidy()` fails for redcaps where the first instrument is repeating and the user specifies `forms` that don't include the first instrument.

# Proposed Changes 

- Update `read_redcap_tidy()` to:
    - remove extra rows added to the data due to the first form being inserted into the `forms` parameter of `redcap_read_oneshot()`
    - remove `redcap_repeat_instrument` and `redcap_repeat_instance` if there were no other repeating forms
- Add test case + mocks to cover this additional situation

### Issue Addressed
Closes #74

### PR Checklist
Before submitting this PR, please check and verify below that the submission meets the below criteria:

- [x] New/revised functions have associated tests
- [x] New/revised functions that update downstream outputs have associated static testing files (`.RDS`) updated under `inst/testdata/create_test_data.R`
- [x] New tests that make API calls use `httptest::with_mock_api` and any new mocks were added to `tests/testthat/fixtures/create_httptest_mocks.R`
- [x] New/revised functions use appropriate naming conventions
- [x] New/revised functions don't repeat code
- [x] Code changes are less than **250** lines total
- [x] Issues linked to the PR using [GitHub's list of keywords](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue)
- [x] The appropriate reviewer is assigned to the PR
- [x] The appropriate developers are assigned to the PR

# Code Review
This section to be used by the reviewer and developers during Code Review after PR submission

### Code Review Checklist

- [ ] I checked that new files follow naming conventions and are in the right place
- [ ] I checked that documentation is complete, clear, and without typos
- [ ] I added/edited comments to explain "why" not "how"
- [ ] I checked that all new variable and function names follow naming conventions
- [ ] I checked that new tests have been written for key business logic and/or bugs that this PR fixes
- [ ] I checked that new tests address important edge cases

### CODE REVIEW

File: R/read_redcap_tidy.R [hunk 1]
```diff
@@ -157,12 +160,27 @@ read_redcap_tidy <- function(redcap_uri,
...
+    # update_data_col_names
     db_data <- select(db_data, !any_of(extra_fields_to_drop))
+
+    # Drop repeating instrument rows that may be associated with extra form
+    if ("redcap_repeat_instrument" %in% colnames(db_data)) {
+      db_data <- db_data %>%
+        filter(.data$redcap_repeat_instrument %in% forms |
+          is.na(.data$redcap_repeat_instrument))
+
+      # Drop repeating fields if there are no remaining repeating instruments
```
Comment (rsh52) [inline R/read_redcap_tidy.R:178]: Nice catch here

**Review by rsh52 — APPROVED** [review]
Looks good to me, works well and the test looks like everything we want.
